cookbooks/aws-parallelcluster-environment/recipes/init/config_default_user_home.rb

```diff
@@ -54,8 +59,30 @@
     rsync -a /tmp#{node['cluster']['cluster_user_home']}/ #{node['cluster']['cluster_user_local_home']}
     usermod -d #{node['cluster']['cluster_user_local_home']} #{node['cluster']['cluster_user']}
     chown -R #{node['cluster']['cluster_user']}: #{node['cluster']['cluster_user_local_home']}
-    rm -rf /tmp#{node['cluster']['cluster_user_home']}
-    rm -rf #{node['cluster']['cluster_user_home']}
+  EOH
+end
+
+# Data integrity check
+# 1. Verifies data integrity by comparing the temporary backup directory and the new local home directory.
+# 2. If the data integrity check passes, it removes both the temporary backup directory and the original home directory.
+# 3. If the data integrity check fails, it outputs an error message and exits with an error code 1.
+# 4. Unhappy Path Manually test passed successfully:
+#    Simulate a failure in the data integrity check by creating expected `node['cluster']['cluster_user_local_home']`
+#    Modify files in the expected directory to differ from the original.
+#    Manually run `config_default_user_home` recipe.
+#    Verified that the script fails and outputs the correct error message in chef-client.log.
+bash "Verify data integrity for #{node['cluster']['cluster_user_home']}" do
+  user 'root'
+  group 'root'
+  code <<-EOH
+    diff_output=$(diff -r #{node['cluster']['cluster_user_home']} #{node['cluster']['cluster_user_local_home']})
+    if [ $? -eq 0 ]; then
+      rm -rf /tmp#{node['cluster']['cluster_user_home']}
+      rm -rf #{node['cluster']['cluster_user_home']}
+    else
+      echo "Data integrity check failed comparing #{node['cluster']['cluster_user_local_home']} and #{node['cluster']['cluster_user_home']}: $diff_output" >&2
+      exit 1
```

Comment:
In previous revision you used to restart sshd because if the integrity check fails, sshd is not restarted by the resource at line 92. Now that it's removed, how can a user login in case of failure?

Reply:
You said we do not restart it in case of failures. So I removed it. For now, customers can login by run this command in terminal:
```
aws ssm send-command \
    --instance-ids "i-0c806e61b7db08afe" \
    --document-name "AWS-RunShellScript" \
    --comment "Start SSHD service" \
    --parameters commands="sudo systemctl start sshd"
```
Or we can add back `systemctl start sshd` in the bash script. It has been tested and worked.

Reply:
> You said we do not restart it in case of failures.

I think there was a misunderstanding. I was explaining why we do stop at the beginning of the recipe, I didn't mean that we should not restart :-) 

Anyway, let's keep the restart

Reply:
Done, added back.